src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,17 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = [];
+
+const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.push(action.payload);
+    },
+    removeBook: (state, action) => state.filter((book) => book.id !== action.payload.id),
+  },
```

Comment:
- [Optional] Kindly `filter()` method does not mutate the original array but instead returns a new array populated with elements of filtered books, so here you are not assigning the returned value to the book state, please assign it to not lose what you removed 👍 
`state.books = state.filter((book) => book.id !== action.payload.id)`

Reply:
Thank you @binhussen; I will incorporate the changes!

Reply:
Hi @binhussen, I have tried to work on your suggestion if you have any feedback please let me know 🙏